shared/fs/banner/public-reminder.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import openUrl from '../../util/open-url'
+
+type Props = {|
+  path: Types.Path,
+|}
+type State = {|
+  closed: boolean,
+|}
+
+const getURL = (path: Types.Path) => {
+  const parsedPath = Constants.parsePath(path)
+  switch (parsedPath.kind) {
+    case 'group-tlf':
+    case 'in-group-tlf':
+      return `https://keybase.pub/${parsedPath.tlfName}`
+    default:
+      return ''
+  }
+}
+
+class PublicBanner extends React.Component<Props, State> {
+  state = {closed: false}
+
+  _onClose = () => this.setState({closed: true})
+
+  render() {
+    if (this.state.closed) {
+      return null
+    }
+    const url = getURL(this.props.path)
+    return (
+      <Kb.Banner
+        color="yellow"
+        text="Everything you upload in here can be viewed by everyone at "
+        actions={[{onClick: () => openUrl(url), title: url + '.'}]}
```

Comment:
If you do [the thing that determines if the banner should show up] inside this component, you'd never need to worry about showing his while url is `''`. I don't feel very strongly about it though so up to you!